src/Gateway/PayplugCreditCard.php

```diff
@@ -248,4 +247,111 @@ public function payment_fields()
 			$this->saved_payment_methods();
 		}
 	}
+
+	/**
+	 * Process the subscription scheduled payment
+	 */
+	public function scheduled_subscription_payment($amount, $order) {
+
+		$order_id      = PayplugWoocommerceHelper::is_pre_30() ? $order->id : $order->get_id();
+		$customer_id = PayplugWoocommerceHelper::is_pre_30() ? $order->customer_user : $order->get_customer_id();
+
+		$payplug_meta = wcs_get_subscription($order->get_meta('_subscription_renewal'))->get_parent()->get_meta("_payplug_metadata");
+
+		if (!$payplug_meta ) {
+			PayplugGateway::log('Could not find the payment token or the payment doesn\'t belong to the current user.', 'error');
+			throw new \Exception(__('Invalid payment method.', 'payplug'));
+		}
+
+		// Retrrienve the saved token
+
+		$token = $this->api->payment_retrieve($payplug_meta['transaction_id'])->card->id;
```

Comment:
why do a payment_retrieve to get the card, when you can do it fetching the card_token from the parent order? (we're forcing the saved card, it should be there on the parent order)

Reply:
get_payment_tokens from the order most of the times is returning empty, so I chose to get the token from the payplug api to be more sure it is always there

Reply:
Perform wise this is not good, subscriptions could be processed all at the same time by batch actions, imagine a customer having 100.000 subscriptions, it will punch 100k request for the api to get info about an order. 

We may need @dmurillo-payplug @dan-payplug input here.